frontend/src/pages/library/_viewId.vue

```diff
@@ -57,258 +65,224 @@ import { getMusicGenresApi } from '@jellyfin/sdk/lib/utils/api/music-genres-api'
 import { getStudiosApi } from '@jellyfin/sdk/lib/utils/api/studios-api';
 import { validLibraryTypes } from '@/utils/items';
 import { useRemote, useSnackbar } from '@/composables';
+import type { Filters } from '@/components/Buttons/FilterButton.vue';
 
-export default defineComponent({
-  async setup() {
-    const { params } = useRoute();
-    const remote = useRemote();
-    const collectionInfo =
-      (
-        await remote.sdk.newUserApi(getItemsApi).getItems({
-          userId: remote.auth.currentUserId,
-          ids: [params.viewId]
-        })
-      ).data?.Items?.[0] ?? {};
+const { t } = useI18n();
+const route = useRoute();
+const remote = useRemote();
 
-    return {
-      useSnackbar,
-      collectionInfo
-    };
-  },
-  data() {
-    return {
-      items: [] as BaseItemDto[],
-      itemsCount: 0,
-      loading: false,
-      viewType: '',
-      sortBy: 'SortName',
-      hasFilters: false,
-      isDefaultView: true, // Movie view, not Collection. Music view, not Genres...
-      statusFilter: [] as string[],
-      genresFilter: [] as string[],
-      yearsFilter: [] as string[],
-      ratingsFilter: [] as string[],
-      filterHasSubtitles: false,
-      filterHasTrailer: false,
-      filterHasSpecialFeature: false,
-      filterHasThemeSong: false,
-      filterHasThemeVideo: false,
-      filterIsHd: undefined as boolean | undefined,
-      filterIs4k: undefined as boolean | undefined,
-      filterIs3d: undefined as boolean | undefined
-    };
-  },
-  computed: {
-    noContent(): boolean {
-      return this.loading || !this.itemsCount;
-    },
-    hasViewTypes(): boolean {
-      return (
-        this.collectionInfo.CollectionType !== undefined &&
-        this.collectionInfo.CollectionType !== null &&
-        this.collectionInfo.CollectionType !== 'homevideos'
-      );
-    },
-    isSortable(): boolean {
-      // Not everything is sortable, so depending on what we're showing, we need to hide the sort menu.
-      // Reusing this as "isFilterable" too, since these seem to go hand in hand for now.
-      return ![
-        'MusicArtist',
-        'Actor',
-        'Genre',
-        'MusicGenre',
-        'MusicGenre',
-        'Studio'
-      ].includes(this.viewType);
-    }
-  },
-  watch: {
-    async viewType(): Promise<void> {
-      await this.refreshItems();
-    },
-    async sortBy(): Promise<void> {
-      await this.refreshItems();
-    }
-  },
-  async mounted() {
-    if (
-      this.collectionInfo?.Type &&
-      validLibraryTypes.includes(this.collectionInfo.Type)
-    ) {
-      if (this.collectionInfo.Name) {
-        this.$route.meta.title = this.collectionInfo.Name;
-      }
+const library = ref<BaseItemDto>();
+const loadingLibrary = ref(false);
+watch(library, (lib) => {
+  if (lib && lib.Name && lib.Type && validLibraryTypes.includes(lib.Type)) {
+    route.meta.title = lib.Name;
+  }
+});
```

Comment:
Please, let's try to keep all the concerns together. The order is not that relevant to me, but ideally all definitions (refs or computed), functions and then watchers at the bottom.

Why? Besides looking more clear and easier to find them if we stick to some convention at this, we avoid ``ReferenceError`` when an immediate watcher could be fired before a variable is assigned.

Also, maybe this can be computed too?

Reply:
I'll re-organize like that. I was organizing these by the parts that they impacted.

Not sure this can be computed though since there's an assignment as part of it which means it's a side effect and should be a watch.

Reply:
@aweebs My point is to have a computed ``libName`` for the if checking and then assign it like ``route.meta.title = libName.value``.

Though now that I'm thinking about this... I'm not sure if maybe we don't need the watcher or the computed property at all since lib should already be reactive? Can you check it please?

Reply:
Using just an assignment without a computed value doesn't work. Neither does assigned it to a computed value like so:
```
const title = computed(() => {
  if (
    library.value &&
    library.value.Name &&
    library.value.Type &&
    validLibraryTypes.includes(library.value.Type)
  ) {
    return library.value.Name;
  }
});

route.meta.title = title.value;
```

I think this will need to stay as a watcher

Reply:
I'm going to simplify this too so that it always just assigned the Name, since only libraries should be loaded here I don't think there's any point in checking if it's a valid library type.

Reply:
@aweebs Fine then.

The check could be moved to the middleware level instead, it doesn't hurt really. Doing that in this PR is up to you though.